src/wp-includes/query.php

```diff
@@ -1161,3 +1161,57 @@ function generate_postdata( $post ) {
 
 	return false;
 }
+
+/**
+ * Simplifies the WordPress loop.
+ *
+ * @since 5.x
+ *
+ * @global WP_Query                    $wp_query WordPress query object.
+ * @global WP_Post                     $post     WordPress post object.
+ *
+ * @param WP_Query|Iterator|array|null $iterable WordPress query object or an array of posts.
+ * @return Generator
+ */
+function wp_loop( $iterable = null ) {
+
+	// If no iterable was passed, use the global query
+	if ( null === $iterable ) {
+		$iterable = $GLOBALS['wp_query'];
+	}
+
+	// If the iterable contains a posts property, use that as the iterable
+	if ( is_object( $iterable ) && property_exists( $iterable, 'posts' ) ) {
+		$iterable = $iterable->posts;
+	}
+
+	// If we don't have a valid iterable, return false and trigger message
+	if ( ! is_iterable( $iterable ) ) {
+		/* translators: Data type */
+		_doing_it_wrong( __FUNCTION__, sprintf( __( 'Expected an iterable, received %s instead.' ), gettype( $iterable ) ), '5.x' );
+
+		return false;
```

Comment:
Return empty array here. What if they pass an iterator or an array whose items are not a valid post ID or WP_Post object?